### PR TITLE
Modified driver_radiation_sw to run on CPUs during an OpenACC run.

### PR DESCRIPTION
**Pull request by Pranay-Reddy-Kommera**
This is the 4th PR submitted in a pool of approximately 12 PRs.
The code changes include appropriate role checks and data transfers to/from CPU and GPU. The code changes allow the 'driver_radiation_sw' subroutine to execute on the CPU while any other dynamics/physics schemes are executing on the GPU. These temporary changes are required to keep track of any intermediate bugs originating as part of the Physics porting efforts. As the physics porting efforts progress, these changes will be removed if the data transfers become redundant.

Code Execution Status:

1. The code compiles and executes successfully on CPU.
2. Code compiles successfully with Dynamics on GPU and Physics on CPU. Code is not executed as the arrays/variables are not yet updated appropriately (will result in NaNs or incorrect answers) in the Physics. The code should be working by the end of all the PRs.

### CODE REVIEW

**Review by cenamiller — APPROVED** [review]

**Comment by cponder** [comment]
Code-changes look good from my end.
Also I ran the `7.x/benchmark/30km` case with these PRs
```893 895 896 902 903 904 905 906 907 908 909 910 911```
and didn't see any failure issue.

**Review by cponder — APPROVED** [review]

**Review by raghuraj19 — APPROVED** [review]

File: src/core_atmosphere/physics/mpas_atmphys_driver_radiation_sw.F [hunk 1]
```diff
@@ -395,8 +395,14 @@ subroutine radiation_sw_from_MPAS(configs,mesh,state,time_lev,diag_physics,atm_i
...
- if (mpas_cpl%role_is(ROLE_INTEGRATE) .or. &
-     mpas_cpl%role_is(ROLE_RADIATION)) then
+ if (mpas_cpl%role_includes(ROLE_INTEGRATE)) then
+    !$acc update host(latCell, lonCell, aerosols, skintemp, snow, &
+    !$acc xice, xland, pin, ozmixm, sfc_albedo, sfc_emiss, cldfrac, &
+    !$acc o3clim, m_hybi, m_ps)
+ endif
+
+ if (mpas_cpl%role_includes(ROLE_INTEGRATE) .or. &
+     mpas_cpl%role_includes(ROLE_RADIATION)) then
```
Comment (mgduda) [inline src/core_atmosphere/physics/mpas_atmphys_driver_radiation_sw.F:405]: The original logic here was:
```fortran
 if (mpas_cpl%role_is(ROLE_INTEGRATE) .or. &
     mpas_cpl%role_is(ROLE_RADIATION)) then
```
which ensured that fields were transferred only if MPI tasks were performing an exclusive role (either INTEGRATE or RADIATION). The new logic,
```fortran
 if (mpas_cpl%role_includes(ROLE_INTEGRATE) .or. &
     mpas_cpl%role_includes(ROLE_RADIATION)) then
```
will transfer fields even in cases where MPI tasks are performing both INTEGRATE and RADIATION roles. Is this the intent?

Comment (mgduda) [reply src/core_atmosphere/physics/mpas_atmphys_driver_radiation_sw.F:405]: Browsing through the code, it looks like the pairings used to transfer fields will not even be generated in cases where MPI tasks are performing both INTEGRATE and RADIATION roles; see the [code from mpas_atm_core.F](https://github.com/MPAS-Dev/MPAS-Model/blob/atmosphere/develop-openacc/src/core_atmosphere/mpas_atm_core.F#L99-L105):
```fortran
      if (mpas_cpl % role_is(ROLE_INTEGRATE)) then
         cells_to_rad_handle = MPAS_coupler_create_pairing(mpas_cpl, MPAS_COUPLER_OUTGOING, ROLE_RADIATION, indexToCellID, nCellsSolve)
         cells_from_rad_handle = MPAS_coupler_create_pairing(mpas_cpl, MPAS_COUPLER_INCOMING, ROLE_RADIATION, indexToCellID, nCellsSolve)
      else if (mpas_cpl % role_is(ROLE_RADIATION)) then
         cells_to_rad_handle = MPAS_coupler_create_pairing(mpas_cpl, MPAS_COUPLER_INCOMING, ROLE_INTEGRATE, indexToCellID, nCellsSolve)
         cells_from_rad_handle = MPAS_coupler_create_pairing(mpas_cpl, MPAS_COUPLER_OUTGOING, ROLE_INTEGRATE, indexToCellID, nCellsSolve)
      end if
```

**Comment by Pranay-Reddy-Kommera** [comment]
I have reverted 'role_includes' to 'role_is'. Please review the code.

**Pull request merged**